Sources/AsyncHTTPClient/HTTPClient+StructuredConcurrency.swift

```diff
@@ -0,0 +1,30 @@
+import NIO
+import Logging
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension HTTPClient {
+    /// Start & automatically shut down a new ``HTTPClient``.
+    ///
+    /// This method allows to start & automatically dispose of a ``HTTPClient`` following the principle of Structured Concurrency.
+    /// The ``HTTPClient`` is guaranteed to be shut down upon return, whether `body` throws or not.
+    ///
+    /// This may be particularly useful if you cannot use the shared singleton (``HTTPClient/shared``).
+    public static func withHTTPClient<R: Sendable>(
+        eventLoopGroup: any EventLoopGroup = HTTPClient.defaultEventLoopGroup,
+        configuration: Configuration = Configuration(),
+        backgroundActivityLogger: Logger? = nil,
+        _ body: @escaping @Sendable (HTTPClient) async throws -> R
+    ) async throws -> R {
```

Comment:
```suggestion
    public static func withHTTPClient<Result>(
        eventLoopGroup: any EventLoopGroup = HTTPClient.defaultEventLoopGroup,
        configuration: Configuration = Configuration(),
        backgroundActivityLogger: Logger? = nil,
        isolation: isolated (any Actor)? = #isolation,
        _ body: (HTTPClient) async throws -> sending Result
    ) async throws -> sending Result {
```

Reply:
@FranzBusch isn't this all Swift 6.0? We support 5.9, right?

Reply:
Partially the `@Sendable @escsaping` isn't generally needed on the body. We need those APIs `isolation/sending` things for `6.0` at least otherwise adopters will get warnings/errors.

Reply:
@FranzBusch So what's the recommendation? Add two, one for 5.9 and one for 6.0 using `#if compiler`?

Reply:
@FranzBusch Okay, that's weird. So the API actually changes depending on the compiler version you use to compile...

Reply:
> Without the #isolation you will get errors in Swift 6.

Well. in certain scenarios, yes. All my stuff is 6 language mode and never got an error. For our software we keep it 5.10 compatible at the minute and migrate to `isolated` once we're 6.0+